controller/lib/core/controller.cpp

```diff
@@ -19,65 +19,69 @@ limitations under the License.
-// https://en.wikipedia.org/wiki/Ziegler%E2%80%93Nichols_method
-//
-// Note that Ku and Tu only seem to work with this particular sample time.
 static constexpr Duration PID_SAMPLE_PERIOD = milliseconds(10);
-static constexpr float Ku = 200;
-static constexpr Duration Tu = seconds(1.5f);
-
-// "No overshoot" settings from the Ziegler-Nichols Wikipedia page.  This
-// avoids overpressurizing the patient's lungs.
-static constexpr float Kp = 0.2f * Ku;
-static constexpr float Ki = 0.4f * Ku / Tu.seconds();
-static constexpr float Kd = Ku * Tu.seconds() / 15;
 
 static DebugFloat dbg_setpoint("setpoint", "Setpoint pressure, kPa");
 
+static DebugFloat dbg_kp("kp", "Proportional gain for main loop", 3.5);
+static DebugFloat dbg_ki("ki", "Integral gain for main loop", 1.0);
+static DebugFloat dbg_kd("kd", "Derivitive gain for main loop");
+static DebugFloat dbg_sp("pc_setpoint", "Pressure control setpoint (cmH2O)");
+
 Controller::Controller()
-    : pid_(Kp, Ki, Kd, ProportionalTerm::ON_ERROR,
-           DifferentialTerm::ON_MEASUREMENT,
-           // Increases in the blower fan speed should result in increased
-           // pressure.
-           ControlDirection::DIRECT,
-           // Our output is an 8-bit PWM.
-           /*output_min=*/0.f, /*output_max=*/255.f, PID_SAMPLE_PERIOD) {}
+    : pid_(0.0f, 0.0f, 0.0f, ProportionalTerm::ON_ERROR,
+           DifferentialTerm::ON_MEASUREMENT, 0.f, 1.0f, PID_SAMPLE_PERIOD) {}
 
 Duration Controller::GetLoopPeriod() { return PID_SAMPLE_PERIOD; }
 
 ActuatorsState Controller::Run(Time now, const VentParams &params,
                                const SensorReadings &readings) {
   BlowerSystemState desired_state = fsm_.DesiredState(now, params);
 
-  return {.fan_setpoint_cm_h2o = desired_state.setpoint_pressure.cmH2O(),
-          .expire_valve_state = desired_state.expire_valve_state,
-          .fan_power = ComputeFanPower(now, desired_state, readings),
-          .fan_valve = 1.0f /* not used yet */};
-}
+  ActuatorsState actuator_state;
 
-float Controller::ComputeFanPower(Time now,
-                                  const BlowerSystemState &desired_state,
-                                  const SensorReadings &sensor_readings) {
-  // If the blower is not enabled, immediately shut down the fan.  But for
-  // consistency, we still run the PID iteration above.
-  float output;
+  pid_.SetKP(dbg_kp.Get());
+  pid_.SetKI(dbg_ki.Get());
+  pid_.SetKD(dbg_kd.Get());
+
+  // Not used yet
+  actuator_state.fio2_valve = 0.0f;
+
+  float pressure = cmH2O(readings.patient_pressure_cm_h2o).kPa();
   float setpoint = desired_state.setpoint_pressure.kPa();
-  dbg_setpoint.Set(setpoint);
-
-  if (desired_state.blower_enabled) {
-    output = pid_.Compute(
-        /*time=*/now,
-        /*input=*/cmH2O(sensor_readings.patient_pressure_cm_h2o).kPa(),
-        /*setpoint=*/setpoint);
-  } else {
-    output = 0;
-    pid_.Observe(/*time=*/now,
```

Comment:
If we're not using Observe anymore, then it's fine to delete it and its tests.

Reply:
I'd like to make a bunch of changes to the PID library but will hold off on that for a separate PR.